zebrad/src/commands/start/sync.rs

```diff
@@ -0,0 +1,204 @@
+use color_eyre::Report;
+use eyre::eyre;
+use futures::stream::{FuturesUnordered, StreamExt};
+use std::{collections::HashSet, iter};
+use tower::{Service, ServiceExt};
+use tracing_futures::Instrument;
+use zebra_chain::{block::BlockHeaderHash, types::BlockHeight};
+
+pub struct Syncer<ZN, ZS>
+where
+    ZN: Service<zebra_network::Request>,
+{
+    pub peer_set: ZN,
+    pub state: ZS,
+    pub tip_requests: FuturesUnordered<ZN::Future>,
+    pub block_requests: FuturesUnordered<ZN::Future>,
+    pub block_locator: Vec<BlockHeaderHash>,
```

Comment:
I don't think we ever need to save the block locator, because every time we request new hashes, we're requesting different data with a newly-constructed block locator: when obtaining new tips, we need to be requesting them with request to the current state, and when extending tips, we only specify a single hash.